boot/zephyr/CMakeLists.txt

```diff
@@ -210,9 +210,13 @@ if(CONFIG_MCUBOOT_SERIAL)
     )
 endif()
 
+get_filename_component(CONF_DIR ${CONF_FILE} DIRECTORY)
+
 if(NOT CONFIG_BOOT_SIGNATURE_KEY_FILE STREQUAL "")
   if(IS_ABSOLUTE ${CONFIG_BOOT_SIGNATURE_KEY_FILE})
     set(KEY_FILE ${CONFIG_BOOT_SIGNATURE_KEY_FILE})
+  elseif(EXISTS ${CONF_DIR}/${CONFIG_BOOT_SIGNATURE_KEY_FILE})
```

Comment:
To be absolutely sure that you are not interfering with existing functionality, but instead extending, you should consider moving this check below the check from the base of the repository.

Reply:
To be honest I prefer to check application configuration directory first. I think that dedicated application keys should have precedence (since they are dedicated for the given application). The default keys should be used only if no other keys are specified.

Reply:
Fine with me :)